Modified clean & concise solution with One Loop

vigneshshiv opened the pull request:
Clean and concise solution with one loop. Please check if this is good to merge..

wingkwong commented on solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md:
```diff
@@ -45,35 +45,32 @@ Notice that the answer must be a substring, "pwke" is a subsequence and not a su
+Move right (j++) till distinct characters and store them in set.
+If repeated character occurs then move left (i++) until that repeated character is occured in left ans also remove all characters that occur before that character including character itself from set.
+
+Time complexity: O(n), where n - # of characters in the string
+Space complexity: O(s), where s is the longest substring
```
- Apply LaTex. 
- Use O instead of 0.
- Line break

vigneshshiv replied:
Modified the file and updated with the requested changes. Please verify once.

wingkwong reviewed (changes requested):
1. Run it locally to verify it first. 
![image](https://user-images.githubusercontent.com/35857179/192923134-45f1e8cd-2ec0-4122-9cd1-5d305bfff949.png)
2. Don't override others' solution. Add it below and point out how to optimize the above solution instead.

vigneshshiv commented:
As suggested, I have modified the file and added another approach, Please take a look if it's good to go. 

The solutions are well tested and accepted in Leetcode, sample test cases found in my repo for the same problem. 
[Link for Testcase solution](https://github.com/vigneshshiv/leetcode/blob/main/code/java/sliding_window/LongestSubstringWithoutRepeatingChars.java)

wingkwong commented on solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md:
```diff
@@ -78,4 +116,44 @@ class Solution {
+</Tabs>
+
+
+## Approach 2: Sliding Window with ASCII
```
It should be Approach 3

vigneshshiv replied:
Updated and pushed it.

wingkwong reviewed (changes requested):
also add your name in Author in L3

vigneshshiv commented:
Updated with last suggestion, please take a look if this is Ok
